[23기 김명성] MODIFY : 수정사항 반영 / ADD : 슬라이드 기능 구현, Nav 추가

sstaar91 opened the pull request:
## 추가 및 수정 사항 간략한 한줄 요약
- main page 검색창 레이아웃 (1d242d7)
- main page 전체 레이아웃 구현 (1e06986)

08/04
- (commit : f68ae51)
   MODIFY : main page 수정사항 반영
   ADD : 슬라이드 기능 구현

08/05
- (commit : 9c46fdf)
   MODIFY : main page 수정사항 반영
   ADD : Nav 추가

08/06
- (commit : abb6a7a)
   REFACTORING : scss 수정, 폴더 위치 수정
   ADD : 검색창 포커스 기능 구현

08/07
- (commit : f6a7d4e)
   MODIFY : 수정 사항 반영

## 추가 및 수정 사항들 자세한 내용
- 검색창 레이아웃 : 글자 30자 제한 / button 중첩 구현
- 전체 레이아웃 : 맛집 리스트, 잇딜 리스트 mock data 이용해 구현

08/04
- @mixin 에 적용하는 속성 수정 (기본값 설정, 중복 설정 생략, 속성 순서 재배치)
- 버튼 클릭시 화면에 안보이는 사진 목록이 보일 수 있도록 슬라이드 기능 구현

08/05
- 네이밍 오류 수정 (직관적일 수 있도록 변경) / 구조분해 할당 적용 / 불필요한 scss import 삭제
- 공용 Nav 코드 이용해 메인 페이지 전용 Nav 구현

08/06
- scss 파일에 스타일 순서 재배치 / 폴더 구조 재배치 / 컴포넌트 분리
- 검색창 클릭시 모달 효과처럼 구현

## 기타 질문 및 특이 사항
- Footer는 정일님께서 만드신 컴포넌트를 이용할 예정입니다.

## 체크 리스트 (아래 사항들이 전부 체크되어야만 merge가 됩니다!)
- [x] 필요한 test들을 완료하였고 기능이 제대로 실행되는지 확인 하였습니다.
- [x] Wecode의 코드 스타일 가이드에 맞추어 코드를 작성 하였습니다.
- [x] 제가 의도한 파일들과 수정 사항들만 커밋이 된 것을 확인 하였습니다.
- [x] 본 수정 사항들을 팀원들과 사전에 상의하였고 팀원들 모두 해당 PR에 대하여 알고 있습니다.

Commit:
ADD : Main page 검색창 레이아웃
- input을 이용해 검색창 제작
- button을 input과 중첩해 제작
- 메인 이미지는 임의 이미지
button

Commit:
ADD : Main Page 레이아웃 구현
- 맛집 리스트 레이아웃 구현
- 잇딜 레이아웃 구현

Tonyk0901 commented on src/pages/Main/style/variable.scss:
```diff
@@ -0,0 +1,31 @@
+  flex-direction: $flex-direction;
+  align-content: $align-content;
+  justify-content: $justify-content;
+}
```
- 너무 잘 해주셨습니다 ✨  만!
- display 부분은 필요가 없어 보입니다. 왜냐면 아래 3가지 속성의 경우 flex 디스플레이일 때만 적용이 되는 속성 입니다. 따라서 sort 를 사용하실때는 무조건 display 가 무조건 flex 로 고정이 될 것 같습니다.
```suggestion
@mixin sort($flex-direction, $align-content, $justify-content) {
  display: flex;
  flex-direction: $flex-direction;
  align-content: $align-content;
  justify-content: $justify-content;
}
```
- 두번째로, 아무런 값이 주어지지 않았을 때 초기 값을 할당해 주시면 좋을 것 같습니다. 예를 들어, justify-content 의 경우 center 가 자주 사용되는데, 이런 부분을 계속 작성해주지 않아도 되게 말이죠!
```suggestion
@mixin sort($flex-direction = row, $align-content = center, $justify-content = center) {
  display: flex;
  flex-direction: $flex-direction;
  align-content: $align-content;
  justify-content: $justify-content;
}

div {
  @include sort();
}

// div 에는 다음과 같이 속성이 부여됩니다.
// 1. display: flex;
// 2. flex-direction: row;
// 3. align-content: center;
// 4. justify-content: center;
```
- 세번째로, flex-direction 은 자주 수정되는 속성은 아니기 때문에 가장 뒤에서 인자로 받도록 설정 하지 않는 것을 추천드립니다. 만약 지금과 같이 앞쪽에 작성해주신 경우, justify-content 를 수정하기 위해서 매번 flex-direction 를 명시를 해주어야만 하기 때문입니다! 
```suggestion
@mixin sort($align-content = center, $justify-content = center, $flex-direction = row) {
  display: flex;
  flex-direction: $flex-direction;
  align-content: $align-content;
  justify-content: $justify-content;
}
```
- 네번째로, 자주 수정이 되는 인자를 앞쪽으로 위치 시켜주시는 게 더 좋습니다. justify-content 가 좀 더 자주 수정이 되기 때문에, 앞쪽으로 위치시켜 주시는게 더 좋아보입니다~
```suggestion
@mixin sort($justify-content = center, $align-content = center, $flex-direction = row) {
  display: flex;
  flex-direction: $flex-direction;
  justify-content: $justify-content;
  align-content: $align-content;
}
```

sstaar91 replied:
성훈님께서 말씀해주신 부분에 힌트를 얻어 적용이 되어있지 않은 부분도 함께 적용해 수정했습니다! 감사합니다

Tonyk0901 commented on src/pages/Main/ShopListContainer/ShopListContainer.scss:
```diff
@@ -0,0 +1,38 @@
+@import '../style/variable.scss';
+
+.shopListContainer {
+  border-bottom: 1px solid rgba(100, 100, 100, 0.2);
+  padding: 40px 7%;
+  .containerTitle {
+    color: #f97100;
+    font-size: 25px;
+  }
+  .filterList {
+    display: flex;
+    justify-content: space-between;
+    flex-wrap: wrap;
+    margin-top: 30px;
+    padding: 10px 20px;
+    .listInfo {
+      @include size(23em, 12em);
+      @include sort(flex, column, space-between, inherit);
+      position: relative;
+      padding-bottom: 20px;
+      margin-bottom: 20px;
+      overflow: hidden;
+      cursor: pointer;
+      .listBackground {
+        z-index: -1;
+        object-fit: cover;
+      }
+      .listTilte {
+        @include contentPosition(absolute, 40%, inherit, inherit, 13%);
+        @include imageTopText(30px);
+      }
+      .listSubTitle {
+        @include contentPosition(absolute, 60%, inherit, inherit, 25%);
+        @include imageTopText(17px);
+      }
+    }
+  }
+}
```
```suggestion
.shopListContainer {
  border-bottom: 1px solid rgba(100, 100, 100, 0.2);
  padding: 40px 7%;
  
  .containerTitle {
    color: #f97100;
    font-size: 25px;
  }
  
  .filterList {
    display: flex;
    justify-content: space-between;
    flex-wrap: wrap;
    margin-top: 30px;
    padding: 10px 20px;
    
    .listInfo {
      @include size(23em, 12em);
      @include sort(flex, column, space-between, inherit);
      position: relative;
      padding-bottom: 20px;
      margin-bottom: 20px;
      overflow: hidden;
      cursor: pointer;
      
      .listBackground {
        z-index: -1;
        object-fit: cover;
      }
      
      .listTilte {
        @include contentPosition(absolute, 40%, inherit, inherit, 13%);
        @include imageTopText(30px);
      }
      
      .listSubTitle {
        @include contentPosition(absolute, 60%, inherit, inherit, 25%);
        @include imageTopText(17px);
      }
    }
  }
}
```

Tonyk0901 commented on src/pages/Main/Main.js:
```diff
@@ -1,10 +1,34 @@
+            <h2>탱고플레이트</h2>
+            <form className="searchForm">
+              <FontAwesomeIcon icon={faSearch} className="searchIcon" />
+              <input
+                type="text"
+                className="searchInput"
+                maxLength="30"
+                placeholder="맛집"
+              ></input>
```
- 컨텐츠가 없다면, 셀프 클로징 해주세요!
```suggestion
              <input
                type="text"
                className="searchInput"
                maxLength="30"
                placeholder="맛집"
              />
```

sstaar91 replied:
넵 수정했습니다👍

Tonyk0901 commented on src/pages/Main/Main.js:
```diff
@@ -1,10 +1,34 @@
   render() {
     return (
       <div className="Main">
-        <div>Main 입니다</div>
+        <header className="MainHeader">
+          <div className="MainTitle">
```
- 소문자로 시작해주세요! 대문자로 시작하는 건 컴포넌트를 네이밍하실 때만 입니다!
```suggestion
<div className='main'>
  <header className='mainHeader'>
    <div className='mainTitle'>
```

sstaar91 replied:
넵 수정했습니다👍

Tonyk0901 commented on src/pages/Main/Main.scss:
```diff
@@ -1,3 +1,51 @@
+    display: flex;
+    flex-direction: column;
+    align-items: center;
+    transform: translate(-50%, -50%);
+    width: 700px;
+    h2 {
+      margin-top: 15px;
+      color: white;
+      font-size: 35px;
+    }
+    .searchForm {
+      position: relative;
+      width: 100%;
+      .searchIcon {
+        @include contentPosition(absolute, inherit, inherit, 12px, 40px);
+        @include size(25px, 25px);
+        color: rgba(196, 196, 196, 0.8);
+      }
+      .searchInput {
+        @include border(2px solid #f97100, 30px);
+        margin-top: 30px;
+        padding: 10px 200px 10px 90px;
+        width: 100%;
+        background-color: white;
+        font-size: 22px;
+        &::placeholder {
+          color: rgba(196, 196, 196, 0.8);
+          font-size: 18px;
+        }
+      }
+      .searchButton {
+        @include border(1px solid #f97100, 30px);
+        @include contentPosition(absolute, inherit, 0, 0, inherit);
+        padding: 10px 70px;
+        background-color: #f97100;
+        color: white;
+        font-size: 22px;
+      }
+    }
+  }
```
```suggestion
.MainHeader {
  @include size(100%, 500px);
  position: relative;
  background-image: url('../Main/images/ex_food.jpg');
  background-size: 1600px;
  
  .MainTitle {
    @include contentPosition(relative, 50%, 0, 0, 50%);
    display: flex;
    flex-direction: column;
    align-items: center;
    transform: translate(-50%, -50%);
    width: 700px;
    
    h2 {
      margin-top: 15px;
      color: white;
      font-size: 35px;
    }
    
    .searchForm {
      position: relative;
      width: 100%;
      
      .searchIcon {
        @include contentPosition(absolute, inherit, inherit, 12px, 40px);
        @include size(25px, 25px);
        color: rgba(196, 196, 196, 0.8);
      }
      
      .searchInput {
        @include border(2px solid #f97100, 30px);
        margin-top: 30px;
        padding: 10px 200px 10px 90px;
        width: 100%;
        background-color: white;
        font-size: 22px;
        
        &::placeholder {
          color: rgba(196, 196, 196, 0.8);
          font-size: 18px;
        }
      }
      
      .searchButton {
        @include border(1px solid #f97100, 30px);
        @include contentPosition(absolute, inherit, 0, 0, inherit);
        padding: 10px 70px;
        background-color: #f97100;
        color: white;
        font-size: 22px;
      }
    }
  }
```

sstaar91 replied:
아무런 코멘트가 없는 부분은 각 적용된 속성마다 가독성을 위해 한칸 씩 떼도록 수정하는 것으로 이해했는데
맞는 수정 방법인가요?

Tonyk0901 replied:
넵-!

Tonyk0901 reviewed (commented):
고생하셨습니다 명성님-! 코드 제 스타일이네요 ✨  크게 수정이 필요한 부분은 없는 것 같습니다! 코멘트 달아드린 것 확인해 주세요~

Commit:
MODIFY : 수정사항 반영
- variable.scss에서 @mixin 수정 사항 반영
- scss 파일의 스타일 간 간격을 벌리며 가독성 확보

ADD : 슬라이드 기능 구현
- ShopListContainer.js 에 포함된 리스트에 슬라이드 기능 구현

Tonyk0901 commented on src/pages/Main/Main.scss:
```diff
@@ -1,3 +1,57 @@
-* {
-  width: 100%;
+@import './style/variable.scss';
+@import './ShopListContainer/ShopListContainer.scss';
```
- ShopListContainer.scss 는 어떤 이유에서 import 하고 계신건가요? 딱히 목적이 없어 보여서 질문 드립니다!

sstaar91 replied:
엇.. 컴포넌트 분리를 하다보니 잠시 착각한 것 같습니다.!
찾아주셔서 감사합니다. 수정했습니다! 👍

Tonyk0901 commented on src/pages/Main/ShopListContainer/ShopListContainer.js:
```diff
@@ -0,0 +1,49 @@
+        <div className="rowContainer">
+          <div className={listPage ? 'filterListLeft' : 'filterListRight'}>
+            {listData.map((data, idx) => {
+              return <ListInfo Info={data} key={idx} />;
```
- Info -> info
```suggestion
              return <ListInfo info={data} key={idx} />;
```

sstaar91 replied:
넵 수정했습니다 👍

Tonyk0901 commented on src/pages/Main/ShopListContainer/ListInfo/ListInfo.js:
```diff
@@ -0,0 +1,19 @@
+class ListInfo extends React.Component {
+  render() {
+    return (
+      <div className="listInfo" key={this.props.Info.id}>
+        <img
+          src={this.props.Info.img}
+          alt="image_file"
+          className="listBackground"
+        />
+        <span className="listTilte">{this.props.Info.title}</span>
+        <span className="listSubTitle">{this.props.Info.subTitle}</span>
```
- 구조분해할당 해주세요!

sstaar91 replied:
넵 수정했습니다! 👍

Tonyk0901 commented on src/pages/Main/ShopListContainer/ShopListContainer.scss:
```diff
@@ -0,0 +1,40 @@
+@import '../style/variable.scss';
+@import './ListInfo/ListInfo.scss';
+@import './SlideButton/SlideButton.scss';
```
- 여기도 ListInfo.scss 랑 SlideButton.scss 는 어떤 이유에서 import 하고 계신걸까요?

sstaar91 replied:
이곳도 역시 정신이 없었던 것 같습니다ㅜㅜ
찾아주셔서 감사합니다. 수정했습니다! 👍

Tonyk0901 reviewed (commented):
코멘트 확인해주세요 명성님~

Tonyk0901 commented on src/pages/Main/ShopListContainer/ShopListContainer.js:
```diff
@@ -0,0 +1,49 @@
+    super();
+    this.state = {
+      listData: [],
+      listPage: true,
```
- 추가적으로 listPage 라는 이름은 적합하지 않은 것 같습니다. listData 와 너무 헷갈리는 부분도 있구요!
- 좀 더 명확하게 구분될 수 있는 이름으로 지어주세요! 
- 그리고 boolean 값이기 때문에 질의형으로 지어주세요! ex) `isModalOn`

sstaar91 replied:
처음 보여지는 부분이 리스트의 왼쪽 화면이기 때문에 `isLeftSide` 라는 이름을 지어봤습니다.

Tonyk0901 commented on src/pages/Main/ShopListContainer/ShopListContainer.js:
```diff
@@ -0,0 +1,49 @@
+            })}
+          </div>
+        </div>
+        <SlideButton click={this.moveImg} data={this.state.listPage} />
```
- 여기 data 라는 props 이름도 어색한 것 같습니다. 어떤 데이터 보다는, 어떤 버튼이 나타나야 하는지에 대한 참 / 거짓 값 이니까요! 이것도 좀 더 명확한 이름으로 수정해 주세요!

sstaar91 replied:
처음 보여지는 버튼이 오른쪽 이동 버튼이라 `rightButton` 이라는 이름을 지어봤습니다.

Tonyk0901 reviewed (commented):
2개 코멘트 더 달았습니다! 확인해주세요!

Commit:
MODIFY : 수정 사항 반영
- 구조분해할당 반영
- 네이밍 변환
- 잘못 삽입 된 scss 경로 수정

ADD : Nav바 적용
- 정일님께서 만드신 코드를 바탕으로 메인페이지에 맞게 수정

Commit:
ADD : 검색창 포커스
- 검색창에 input을 클릭 할 때, 모달창 처럼 나오는 효과 구현

lang92 commented on src/index.js:
```diff
@@ -1,7 +1,7 @@
 import Routes from './Routes';
-import './styles/common.scss';
-import './styles/reset.scss';
+import './pages/styles/common.scss';
+import './pages/styles/reset.scss';
```
-  이렇게 common.scss 이후에 reset.scss를 import해주시면 common.scss 이후에 reset.scss이 적용되게 됩니다.

- styles 폴더는 src 폴더 하위에 위치하도록 하는 것이 위코드 컨벤션입니다.
혹시 styles 폴더를 pages 폴더 내부로 이동하신 이유가 있을까요?

sstaar91 replied:
아 파일을 구성할 때 잘못 구성 한것 같습니다
scss 적용 순서와 폴더 위치 변경하도록 하겠습니다!

lang92 commented on src/pages/Main/Nav/Nav.js:
```diff
@@ -0,0 +1,43 @@
+            </span>
+          </li> */}
+          <div className="myPage">
+            <i className="far fa-user fa-2x"></i>
```
셀프 클로징 태그 적용해주시기 바랍니다. 리뷰하지 않은 부분도 수정해주세요!

lang92 commented on src/pages/Main/Nav/Nav.js:
```diff
@@ -0,0 +1,43 @@
+    return (
+      <nav className="menu">
+        <div className="leftWrapper">
+          <Link to="/main">
```
메인 페이지 경로가 수정된 만큼 이 부분도 바뀌어야 할 것 같네요!

lang92 commented on src/pages/Main/Nav/Nav.scss:
```diff
@@ -0,0 +1,67 @@
+      border-left: 1px solid #cecece;
+
+      span {
+        display: block;
```
span 태그에 block속성을 적용하실 거면, block 속성을 가진 p태그를 사용하시면 될 것 같습니다.

lang92 commented on src/pages/Main/Nav/Nav.scss:
```diff
@@ -0,0 +1,67 @@
+
+    .eatDealMenu {
+      position: relative;
+      img {
```
가독성을 위해 새로운 셀렉터를 시작할 때는 한 줄 띄워주시기 바랍니다!

lang92 commented on src/pages/Main/Nav/Nav.scss:
```diff
@@ -0,0 +1,67 @@
+      @include sort(inherit, center, center);
+      padding: 15px 30px;
+      border-left: 1px solid #cecece;
+      .fa-user {
```
여기도 한 줄 띄워서 써주세요 :)

lang92 commented on src/pages/Main/SearchFocus/SearchFocus.js:
```diff
@@ -0,0 +1,14 @@
+  render() {
+    return (
+      <div
+        className={this.props.modalActive ? 'modlaHidden' : 'searchInputFocus'}
```
오타가 있네요!
```suggestion
        className={this.props.modalActive ? 'modalHidden' : 'searchInputFocus'}
```

sstaar91 replied:
넵 수정했습니다!

lang92 commented on src/pages/Main/ShopListContainer/ShopListContainer.js:
```diff
@@ -0,0 +1,49 @@
+            })}
+          </div>
+        </div>
+        <SlideButton click={this.moveImg} rightButton={this.state.isLeftSide} />
```
이 부분 구조분해할당 빼먹으신 것 같네요!
그리고 rightButton인데 반대되는 isLeftSide라는 명칭과 사용은 혼란을 줄 수 있을 것 같습니다.

```suggestion
        <SlideButton click={this.moveImg} isLeftBtn={isLeftBtn} />
```

sstaar91 replied:
감사합니다.
버튼 컴포넌트도 구조분해 함께 하고
명칭도 수정했습니다!

lang92 commented on src/pages/Main/SlideButton/SlideButton.js:
```diff
@@ -0,0 +1,25 @@
+import './SlideButton.scss';
+
+class SlideButton extends React.Component {
+  render() {
```
구조분해할당 해주시면 더욱 좋을 것 같네요.

lang92 commented on src/pages/Main/SlideButton/SlideButton.js:
```diff
@@ -0,0 +1,25 @@
+  render() {
+    return (
+      <>
+        <div
+          onClick={this.props.click}
+          className={this.props.rightButton ? 'clickRight' : 'clickRightHidden'}
+        >
+          <i className="fas fa-chevron-right"></i>
+        </div>
+        <div
+          onClick={this.props.click}
+          className={this.props.rightButton ? 'clickLeftHidden' : 'clickLeft'}
+        >
+          <i className="fas fa-chevron-left"></i>
+        </div>
```
hidden이 아닌 상태일 때, 리스트 컨테이너의 아랫쪽에 자리를 차지하고 있네요. 이렇게 되면 나중에 레이아웃을 수정할 일이 생기면 유지보수가 어려울 수 있습니다.
또한, 명칭 관련해서 위의 리뷰처럼 혼란을 줄 수 있어 다음과 같이 변경해볼 수 있을 것 같습니다.
```suggestion
        <div
          onClick={click}
          className={`rightBtn ${isLeftBtn ? 'hidden' : ''}`}
        >
          <i className="fas fa-chevron-right" />
        </div>
        <div onClick={click} className={`leftBtn ${isLeftBtn ? '' : 'hidden'}`}>
          <i className="fas fa-chevron-left" />
        </div>
```

sstaar91 replied:
아 감사합니다!
해당 부분을 잠시 잊고 있었는데
덕분에 훨씬 가독성이나 유지보수 측면에서 도움이 됐습니다!

lang92 commented on src/pages/Main/SlideButton/SlideButton.scss:
```diff
@@ -0,0 +1,19 @@
+@import '../style/variable.scss';
+
+.clickRight {
+  @include alignClickBtun;
+  right: 70px;
+}
+
+.clickRightHidden {
+  @include hide;
+}
+
+.clickLeft {
+  @include alignClickBtun;
+  left: 70px;
+}
+
+.clickLeftHidden {
+  @include hide;
+}
```
위 리뷰 참고해서 수정해보시면 될 것 같습니다.

그리고 alignClickBtun 명칭이 조금 어색한 것 같네요.

lang92 commented on src/pages/Main/Main.js:
```diff
@@ -1,10 +1,42 @@
+  constructor() {
+    super();
+    this.state = {
+      isVaild: true,
```
isValid 명칭만 봤을 때, 어떤 용도로 사용하는 boolean state인지 직관적으로 판단할 수가 없을 것 같네요.
isSearchModalActive와 같이 명확하게 네이밍 해주세요!

sstaar91 replied:
넵 조금더 명확한 네이밍을 할 수 있도록 노력해보겠습니다..!

lang92 reviewed (commented):
고생 많으셨습니다 명성님! 코드를 전체적으로 깔끔하게 잘 구현해주고 계시네요!
리뷰 잘 반영해주시고, 이대로 남은 기능 구현 잘 해주시면 되겠습니다 🙌

Commit:
MODIFY : 수정사항 반영
